### PR TITLE
remove need for fltables.d

### PR DESCRIPTION
**Pull request by WalterBright**
Now that backend is in D, don't need this kludge to initialize arrays. Use CTFE instead.

### CODE REVIEW

File: src/dmd/backend/cg.d [hunk 1]
```diff
@@ -52,3 +50,110 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
...
+                case FLgot:     segfl[i] = cast(byte)-1;  break;
+                case FLgotoff:  segfl[i] = cast(byte)-1;  break;
+                case FLlocalsize: segfl[i] = cast(byte)-1;        break;
+                case FLtlsdata: segfl[i] = cast(byte)-1;  break;
+                case FLframehandler:    segfl[i] = cast(byte)-1;  break;
+                case FLasm:     segfl[i] = cast(byte)-1;  break;
+                case FLallocatmp:       segfl[i] = SS;  break;
+                case FLfuncarg:         segfl[i] = SS;  break;
+                default:
+                        printf("error in segfl[%d]\n", i);
```
Comment (Geod24) [inline src/dmd/backend/cg.d]: Is that CTFE-able ?

Comment (WalterBright) [reply src/dmd/backend/cg.d]: No, but that code is unreachable.

File: src/dmd/backend/cg.d [hunk 2]
```diff
@@ -52,3 +50,110 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
 int TARGET_STACKALIGN = 2;      // default for 16 bit code
 int STACKALIGN = 2;             // varies for each function
 }
+
+import core.stdc.stdio;
+
+extern (C++) __gshared
```
Comment (Geod24) [inline src/dmd/backend/cg.d]: Does it have to be `extern(C++)` ?

File: src/dmd/backend/cg.d [hunk 3]
```diff
@@ -52,3 +50,110 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
...
+        ubyte[FLMAX] segfl;
+
+        /* Segment registers    */
+        enum ES = 0;
+        enum CS = 1;
+        enum SS = 2;
+        enum DS = 3;
+
+        foreach (i;  0 .. FLMAX)
+        {   switch (i)
```
Comment (Geod24) [inline src/dmd/backend/cg.d]:
```suggestion
        {
            switch (i)
```

File: src/dmd/backend/cg.d [hunk 4]
```diff
@@ -52,3 +50,110 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
...
+            {
+                case 0:         segfl[i] = cast(byte)-1;  break;
```
Comment (Geod24) [inline src/dmd/backend/cg.d]:
```suggestion
                case 0:         segfl[i] = byte(-1);  break;
```

Comment (Geod24) [reply src/dmd/backend/cg.d]: And all following occurences

File: src/dmd/backend/cg.d (hunk 1 above)
Comment (Geod24) [reply src/dmd/backend/cg.d]: Then we can just remove the `printf` and the `import`

File: src/dmd/backend/cg.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/backend/cg.d]: No. But as my usual practice, I change one thing at a time, for reasons explained many times.

File: src/dmd/backend/cg.d (hunk 4 above)
Comment (WalterBright) [reply src/dmd/backend/cg.d]: As this is a copy/paste refactoring job, I avoid making more changes in the same PR.

File: src/dmd/backend/cg.d (hunk 3 above)
Comment (WalterBright) [reply src/dmd/backend/cg.d]: It's that way as it's a copy/paste job. Nothing makes reviewing a copy/paste job harder than succumbing to rewriting it in the same PR.

**Comment by WalterBright** [comment]
> Then we can just remove the printf and the import

As a follow-on, perhaps. But one thing at a time.

I'll pull my experience card here. I've done a lot of this sort of copy/paste refactoring. Sometimes I've "fixed" it at the same time. Simple, obvious things. Then, weird errors start happening. I spend lots of time tracking down some innocuous mistake. Once again, I learn the lesson. When copy/pasting, do not attempt to fix the code. Just copy/paste it. Get it all working. Merge it. *THEN* go about improving it.

It also makes bisecting to find where the code broke MUCH MUCH MUCH easier.

**Comment by WalterBright** [comment]
P.S. It's also why this PR only does one of the files optabgen generates.

**Comment by Geod24** [comment]
@WalterBright : Moving code is not very reviewable on `git` anyway. Most of the time, the copying adds bits which don't make much sense, and it does not make reviewing harder because the diff is large. I would much prefer if you fixed up the code before moving it. But let's agree to disagree here, and I'm looking forward to the follow up PRs.

**Pull request merged**